### ISSUE TITLE
$factoryClass - Auth\Source\Factory issue

### ISSUE THREAD

**Issue by kuhnflyfish** [simplesamlphp/simplesamlphp#1468]
# **Help us to keep this issue-tracker clean!  For questions or support, please refer to our [mailing lists](https://simplesamlphp.org/lists)**

**Describe the bug**
include(sspmod_saml_Auth_Source_Factory_SP.php): failed to open stream: No such file or directory

**To Reproduce**
1. Require authentication via $as->requireAuth();
2. This eventually leads to SimpleSAML/Auth/Source.php file.
3. Line 324-328 where variable $factoryClass is defined, it passes the string 'Auth\Source\Factory' as the $type parameter to Module::resolveClass() method.
4. This will create a class called 'SimpleSAML\Module\saml\Auth\Source\Factory\SP' which doesn't exist, so the method tries to then find 'sspmod_saml_Auth_Source_Factory_SP' as the $oldClassName.  What does exist is 'SimpleSAML\Module\saml\Auth\Source\SP' class.  Note the missing 'Factory'.
5. So, by removing 'Factory' from the end of line 326 in Source.php, the code works and I am redirected to Azure AD, our IDP.

**Expected behavior**
If Module::resolveClass() is throwing an exception, why is it not being caught in Source.php on lines 333-337 where it would resolve this 'Factory' issue?

**Screenshots or logs**
<img width="629" alt="Screen Shot 2021-06-02 at 1 31 43 PM" src="https://user-images.githubusercontent.com/42548262/120543695-d546ac00-c3a9-11eb-8fa4-81784289efd3.png">

**Additional context**
This is being done within a Yii 1.1 framework

**Comment by tvdijen** [comment]
`include()` doesn't throw an exception, but emits a warning if the file is not found..
Using `require()` will make it throw a CompileError, which we can catch and deal with...

 I'm not entirely sure where we call this `include()` though.. I'm worried it's in the autoloader-code and changing that may have unexpected side-effects..

**Comment by davidbuzz** [comment]
i hit this issue

**Comment by dmlang** [comment]
Hi, is there a change that this bug will be resolved soon?

**Comment by tvdijen** [comment]
If you can help me reproduce this..  The steps described by the original poster is not reproducible.. i.e. the `$oldClassName` variable never existed in any previous version... Also, I have many SPs running like this without any issue.. Can you describe your situation, starting with the version of SSP being used?

**Comment by dmlang** [comment]
> If you can help me reproduce this.. The steps described by the original poster is not reproducible.. i.e. the `$oldClassName` variable never existed in any previous version... Also, I have many SPs running like this without any issue.. Can you describe your situation, starting with the version of SSP being used?

Hey, thanks for the fast response! I just tried to narrow down the problem and maybe found a problem. I installed simplesamlphp via composer (v. 1.19.3). However, the code of class SimpleSAML/Module is not the same as downloaded by composer. I checked it twice, on different machines. Is there probably a problem with the composer package?

**Comment by tvdijen** [comment]
Maybe you're looking at the `master`-branch.. That one has diverged substantially from the `simplesamlphp-1.19` branch..
How exactly are you using the SP source?

**Comment by dmlang** [comment]
yeah your right, didn't recognize the different version branches... The Module Class code of the master-branch would solve the problem.

> How exactly are you using the SP source?

I just started playing around with the lib:

`    $auth = new Simple('default-sp');

        if (!$auth->isAuthenticated()) {

            Session::getSessionFromRequest()->cleanup();
            $auth->requireAuth([
                'ReturnTo' => 'URL to return to',
                'KeepPost' => FALSE,
            ]);

        } else {
            $attrs = $auth->getAttributes();
            var_dump($attrs);
        }`

The config file is very near to the template provided. I just changed the idp attribute in the authsources file. 
Authenticating via the federation tab of the admintool works just fine. Do you like me to provide more/different information?

**Comment by tvdijen** [comment]
No, this is good! It gives me enough to try and reproduce it...
I mean, I can't remove the broken code in a bugfix-release, because that would impose a backwards compatible break.. However, I can try to fix the original intent of it.

**Comment by dmlang** [comment]
> No, this is good! It gives me enough to try and reproduce it... I mean, I can't remove the broken code in a bugfix-release, because that would impose a backwards compatible break.. However, I can try to fix the original intent of it.

Hi, sry for being impatient, but is there probably an update on this topic? I need the composer package for my deployment process, so hot fixing it does not work for me :(.

**Comment by tvdijen** [comment]
I don't expect to have time for this before the end of the year, sorry..

**Comment by dmlang** [comment]
Hi, no problem. Thanks for the info.

**Comment by dmlang** [comment]
Hi, any news on this issue? thanks.

**Comment by tvdijen** [comment]
@dmlang I tried to reproduce this with 1.19.6 using your code example but it works fine..
Is this still an issue?
